File: internal/handler/risk_handler.go
package handler

import (
	"encoding/json"
	"net/http"

	"github.com/securerent/risk-assessment-service/internal/model"
	"github.com/securerent/risk-assessment-service/internal/service"
)

type RiskHandler struct {
	riskService service.RiskService
}

func NewRiskHandler(rs service.RiskService) *RiskHandler {
	return &RiskHandler{riskService: rs}
}

func (h *RiskHandler) Assess(w http.ResponseWriter, r *http.Request) {
	if r.Method != http.MethodPost {
		http.Error(w, "Method not allowed", http.StatusMethodNotAllowed)
		return
	}

	var profile model.RenterProfile
	if err := json.NewDecoder(r.Body).Decode(&profile); err != nil {
		http.Error(w, "Invalid request body", http.StatusBadRequest)
		return
	}

	result, err := h.riskService.AssessRenter(profile)
	if err != nil {
		http.Error(w, "Failed to assess risk", http.StatusInternalServerError)
		return
	}

	w.Header().Set("Content-Type", "application/json")
	json.NewEncoder(w).Encode(result)
}

func (h *RiskHandler) HealthCheck(w http.ResponseWriter, r *http.Request) {
	w.WriteHeader(http.StatusOK)
	w.Write([]byte(`{"status":"up"}`))
}

File: node/src/handlers/riskHandler.js
import { RiskService } from '../services/riskService.js';

const riskService = new RiskService();

export const assess = (req, res) => {
    try {
        const profile = req.body;

        // Basic validation
        if (!profile.annual_income || !profile.credit_score || !profile.monthly_rent) {
            return res.status(400).json({ error: "Missing required fields" });
        }

        const result = riskService.assessRenter(profile);
        res.json(result);
    } catch (error) {
        res.status(500).json({ error: "Internal server error during assessment" });
    }
};

export const healthCheck = (req, res) => {
    res.json({ status: "up" });
};
